ci/dev/vscode.patch

```diff
@@ -696,9 +696,52 @@ index 2185bb5228c..35463ca6520 100644
  			(err: any, socket: ISocket | undefined) => {
  				if (err || !socket) {
  					options.logService.error(`${logPrefix} socketFactory.connect() failed. Error:`);
+diff --git a/src/vs/platform/storage/browser/storageService.ts b/src/vs/platform/storage/browser/storageService.ts
+index 59b1baf912..cf9805554b 100644
+--- a/src/vs/platform/storage/browser/storageService.ts
++++ b/src/vs/platform/storage/browser/storageService.ts
+@@ -116,8 +116,8 @@ export class BrowserStorageService extends Disposable implements IStorageService
+ 		return this.getStorage(scope).getNumber(key, fallbackValue);
+ 	}
+ 
+-	store(key: string, value: string | boolean | number | undefined | null, scope: StorageScope): void {
+-		this.getStorage(scope).set(key, value);
++	store(key: string, value: string | boolean | number | undefined | null, scope: StorageScope): Promise<void> {
++		return this.getStorage(scope).set(key, value);
+ 	}
+ 
+ 	remove(key: string, scope: StorageScope): void {
+diff --git a/src/vs/platform/storage/common/storage.ts b/src/vs/platform/storage/common/storage.ts
+index 1623957cb1..d366438d54 100644
+--- a/src/vs/platform/storage/common/storage.ts
++++ b/src/vs/platform/storage/common/storage.ts
+@@ -83,7 +83,7 @@ export interface IStorageService {
+ 	 * The scope argument allows to define the scope of the storage
+ 	 * operation to either the current workspace only or all workspaces.
+ 	 */
+-	store(key: string, value: string | boolean | number | undefined | null, scope: StorageScope): void;
++	store(key: string, value: string | boolean | number | undefined | null, scope: StorageScope): Promise<void> | void;
```

Comment:
Why `| void`?

Reply:
There are a bunch of tests that implement this interface. We could modify them all to return promises instead of void but it would increase the patch size a decent amount.